packages/apple-pay-integration/src/ApplePayPaymentMethod.tsx

```diff
@@ -0,0 +1,43 @@
+import { toResolvableComponent, PaymentMethodProps } from '@bigcommerce/checkout-js/payment-integration';
+import React, { FunctionComponent, useEffect } from 'react';
+
+const ApplePaymentMethod: FunctionComponent<PaymentMethodProps> = ({ method, checkoutService, language, onUnhandledError }) => {
+    useEffect(() => {
+        const componentDidMount = async ():Promise<void> => {
+            try {
+                await checkoutService.initializePayment({
+                    gatewayId: method.gateway,
+                    methodId: method.id,
+                    applepay: {
+                        shippingLabel: language.translate('cart.shipping_text'),
+                        subtotalLabel: language.translate('cart.subtotal_text'),
+                    },
+                });
+            } catch (error) {
+                onUnhandledError(error);
+            }
+        }
+        componentDidMount().then();
+
+        return () => {
+            const componentWillUnmount = async ():Promise<void> => {
+                try {
+                    await checkoutService.deinitializePayment({
+                        gatewayId: method.gateway,
+                        methodId: method.id,
+                    });
+                } catch (error) {
+                    onUnhandledError(error);
+                }
+            }
+            componentWillUnmount().then();
+        }
+    }, [checkoutService, language, method, onUnhandledError]);
+
+    return <React.Fragment />;
```

Comment:
Should this return something other than a fragment?

Reply:
In `<PaymentMethod/>` v1, we simply return `null`.
https://github.com/bigcommerce/checkout-js/blob/42ee27d679f81eaabc8b43771b08575525962d4a/packages/core/src/app/payment/paymentMethod/HostedPaymentMethod.tsx#L110